src/python/grpcio/grpc/_cython/_cygrpc/aio/callback_common.pyx.pxi

```diff
@@ -104,10 +109,67 @@ async def callback_start_batch(GrpcCallWrapper grpc_call_wrapper,
         wrapper.c_functor(), NULL)
 
     if error != GRPC_CALL_OK:
-        raise RuntimeError("Failed grpc_call_start_batch: {}".format(error))
+        raise CallbackStartBatchError("Failed grpc_call_start_batch: {}".format(error))
 
     await future
     cpython.Py_DECREF(wrapper)
     cdef grpc_event c_event
     # Tag.event must be called, otherwise messages won't be parsed from C
     batch_operation_tag.event(c_event)
+
+
+async def _receive_message(GrpcCallWrapper grpc_call_wrapper,
+                           object loop):
+    """Retrives parsed messages from C-Core.
+    
+    The messages maybe already in C-Core's buffer, so there isn't a 1-to-1
+    mapping between this and the underlying "socket.read()". Also, eventually,
+    this function will end with an EOF, which reads empty message.
+    """
+    cdef ReceiveMessageOperation receive_op = ReceiveMessageOperation(_EMPTY_FLAG)
+    cdef tuple ops = (receive_op,)
+    try:
+        await callback_start_batch(grpc_call_wrapper, ops, loop)
+    except CallbackStartBatchError as e:
+        # NOTE(lidiz) The receive message operation has two ways to indicate
+        # finish state : 1) returns empty message due to EOF; 2) fails inside
+        # the callback (e.g. cancelled).
+        #
+        # Since they all indicates finish, they are better be merged.
+        _LOGGER.exception(e)
```

Comment:
I changed the logging level from `exception` to `debug`.

The return value of `grpc_call_start_batch` is accountable for scheduling the operations, and it would fail if the cq is not in correct state, or the call is already cancelled, etc.. To obtain the final state of the operation, we need to wait until the callback is invoked with the `success` flag.

I don't have a better solution than this one, since the raised exception is expected. Logging the actual exception helps debugging, I think lowering it to `debug` level should be more accurate.